File: runtimes/nodejs/src/cli/clex-scan.js
#!/usr/bin/env node

/**
 * Clex-AI-Ultra Core - Rust Validator CLI (scan)
 * clex-scan: scan skills directory for structure validation
 */

const fs = require('fs');
const path = require('path');

function main() {
  const skillsDir = path.join(__dirname, '..', '..', '..', 'skills');
  
  if (!fs.existsSync(skillsDir)) {
    console.log('Error: skills directory not found');
    process.exit(1);
  }
  
  const skills = fs.readdirSync(skillsDir).filter(d => fs.statSync(path.join(skillsDir, d)).isDirectory());
  
  console.log('Scanning skills...');
  
  skills.forEach(skill => {
    const skillPath = path.join(skillsDir, skill);
    const skillMd = path.join(skillPath, 'SKILL.md');
    
    if (fs.existsSync(skillMd)) {
      console.log(`  ✅ ${skill}`);
    } else {
      console.log(`  ⚠️  ${skill} (SKILL.md missing)`);
    }
  });
  
  console.log(`\nTotal: ${skills.length} skills scanned`);
}

main();

File: runtimes/rust/src/bin/clex-scan.rs
extern crate serde;
extern crate serde_json;

use std::fs;
use std::path::Path;

fn main() {
    let skills_dir = std::env::args().nth(1).unwrap_or_else(|| "../skills".to_string());
    
    if !Path::new(&skills_dir).exists() {
        eprintln!("Directory not found: {}", skills_dir);
        std::process::exit(1);
    }
    
    println!("Scanning skills in: {}", skills_dir);
    
    let entries = fs::read_dir(&skills_dir).expect("Failed to read directory");
    
    let mut count = 0;
    for entry in entries {
        let entry = entry.expect("Failed to read entry");
        let path = entry.path();
        
        if path.is_dir() {
            let skill_name = path.file_name().unwrap().to_string_lossy();
            let skill_md = path.join("SKILL.md");
            
            if skill_md.exists() {
                println!("  ✅ {}", skill_name);
            } else {
                println!("  ⚠️  {} (SKILL.md missing)", skill_name);
            }
            count += 1;
        }
    }
    
    println!("\nTotal: {} skills scanned", count);
}
